lib/datadog/core/telemetry/client.rb

```diff
@@ -41,7 +44,16 @@ def disable!
         def started!
           return unless @enabled
 
-          @emitter.request('app-started')
+          res = @emitter.request('app-started')
+
+          # Telemetry is only supported by agent versions 7.34 and up
+          if res.not_found?
+            Datadog.logger.debug('Agent does not support telemetry; disabling future telemetry events.')
+            @enabled = false
```

Comment:
You probably want to call `disable!` here, in order to stop the worker as well.

Reply:
Actually the worker wasn't started yet 🤔

Reply:
The worker starts up after `started!` gets called - to ensure the `app-started` event gets sent before `heartbeat` -- would it be better to initialize the worker inside of `#started!` instead?

Reply:
I think it's ok as is today then.